src/test/java/com/walmartlabs/x12/checksum/Crc16Test.java

```diff
@@ -0,0 +1,92 @@
+package com.walmartlabs.x12.checksum;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class Crc16Test {
+    CyclicRedundancyCheck crcUtil;
+
+    @Before
+    public void init() {
+        crcUtil = new Crc16();
+    }
+
+    @Test
+    public void test_generateCyclicRedundancyCheck_null() {
+        String blockText = null;
+        String crcValue = crcUtil.generateCyclicRedundancyCheck(blockText);
+        assertNull(crcValue);
+    }
+
+    @Test
+    public void test_generateCyclicRedundancyCheck_empty() {
+        String blockText = "";
+        String crcValue = crcUtil.generateCyclicRedundancyCheck(blockText);
+        assertNull(crcValue);
+    }
+
+    @Test
+    public void test_generateCyclicRedundancyCheck_hello_world() {
+        String blockText = "hello\r\nworld";
+        String crcValue = crcUtil.generateCyclicRedundancyCheck(blockText);
+        assertNotNull(crcValue);
+        assertEquals("FC4F", crcValue);
+    }
+
+    @Test
+    public void test_generateCyclicRedundancyCheck_123456789() {
+        String blockText = "123456789";
+        String crcValue = crcUtil.generateCyclicRedundancyCheck(blockText);
+        assertNotNull(crcValue);
+        assertEquals("BB3D", crcValue);
+    }
+
+    @Test
+    public void test_verifyBlockOfText_null() {
+        String crcValue = null;
```

Comment:
Should we add a test case for the `verifyBlockOfText()` method where only the crcValue or blockText is null?